src/main/java/com/commercetools/sync/products/helpers/ProductReferenceResolver.java

```diff
@@ -386,7 +386,7 @@ public ProductReferenceResolver(@Nonnull final ProductSyncOptions productSyncOpt
         final List<CompletionStage<Map<String, String>>> futures = new ArrayList<>();
 
         final Set<String> productKeys = referencedKeys.getProductKeys();
-        if (!productKeys.isEmpty()) {
+        if (productKeys.isEmpty()) {
```

Comment:
why execute the reference fetch when we do not have any keys?

Reply:
no reason, it's typo, reverted: https://github.com/commercetools/commercetools-sync-java/pull/581/commits/5c22820a06e5c339a127aa7a6a5b6d8f30ee7f60